release/docker/build-image-multi-arch.sh

```diff
@@ -0,0 +1,141 @@
+#!/bin/bash
+
+set -e
+
+# Import libs
+. ../../lib/shell/file_management.sh
+
+# Variable
+OLDIFS=$IFS
+BUILDER_NUM=`date +%s`
+BUILDER_NAME="multiarch_${BUILDER_NUM}"
+DIR=""
+
+function usage() {
+    echo ""
+    echo "This script is used to build the OpenSearch Docker image with single architecture (x64 or arm64). It prepares the files required by the Dockerfile in a temporary directory, then builds and tags the Docker image."
+    echo "--------------------------------------------------------------------------"
+    echo "Usage: $0 [args]"
+    echo ""
+    echo "Required arguments:"
+    echo -e "-v VERSION\tSpecify the OpenSearch version number that you are building, e.g. '1.0.0' or '1.0.0-beta1'. This will be used to label the Docker image. If you do not use the '-o' option then this tool will download a public OPENSEARCH release matching this version."
+    echo -e "-f DOCKERFILE\tSpecify the dockerfile full path, e.g. dockerfile/opensearch.al2.dockerfile."
+    echo -e "-p PRODUCT\tSpecify the product, e.g. opensearch or opensearch-dashboards, make sure this is the name of your config folder and the name of your .tgz defined in dockerfile."
+    echo -e "-a ARCHITECTURE\tSpecify the multiple architecture you want to add to the multi-arch image, separate by comma, e.g. 'x64,arm64'."
+    echo -e "-r REPOSITORY\tSpecify the docker repository name in the format of '<Docker Hub RepoName>/<Docker Image Name>:<Tag Name>', due to multi-arch image either save in cache or directly upload to Docker Hub Repo, no local copies."
```

Comment:
If this is really required add in the help message. If not add it in optional args

Reply:
They are all required parameters to run the scripts.

Reply:
The reason being I didn't specify the REPOSITORY and the script still ran. If it is hard core required please add here: https://github.com/opensearch-project/opensearch-build/pull/98/files#diff-b5bd2c1d433b43e933753221e7178a5cb2916e96ece90a632ca2c782eb221128R81

Reply:
If it's required then there should be a check that a value has been provided, same as what you're doing on line 81 for the other required arguments.

Reply:
Missed, added -r.

Reply:
Because this is a multi-node image, you cannot save a local image due to your local only have 1 CPU arch. The buildx is using a running container to build, thus cache it. I tried the cached methods but does not work as expected, so the only option is as soon as it builds, --push to the docker registry where it can store both arch layers.